src/map/packet_system.cpp

```diff
@@ -3487,8 +3487,15 @@ void SmallPacket0x050(map_session_data_t* const PSession, CCharEntity* const PCh
     }
 
     charutils::EquipItem(PChar, slotID, equipSlotID, containerID); // current
-    charutils::SaveCharEquip(PChar);
-    charutils::SaveCharLook(PChar);
+    // Only save char look to db every X seconds (or more) to avoid db spam when changing individual pieces
+    // Do not save equipment at all since SaveCharEquip is a very expensive operation
+    auto lastSwapTime   = PChar->GetLocalVar("core-LastGearSwap");
```

Comment:
Please extract the key `std::string` "core-LastGearSwap" to the top of the file into an anonymouse namespace, like in zone_entities.cpp:
```cpp

namespace
{
    const float CHARACTER_SYNC_DISTANCE                = 45.0f;
    const float CHARACTER_DESPAWN_DISTANCE             = 50.0f;
    const int   CHARACTER_SWAP_MAX                     = 5;
    const int   CHARACTER_SYNC_LIMIT_MAX               = 32;
    const int   CHARACTER_SYNC_DISTANCE_SWAP_THRESHOLD = 30;
    const int   CHARACTER_SYNC_PARTY_SIGNIFICANCE      = 100000;
    const int   CHARACTER_SYNC_ALLI_SIGNIFICANCE       = 10000;
    const int   PERSIST_CHECK_CHARACTERS               = 20;
} // namespace

```

Reply:
Why not use a variable to store this in CCharEntity over using localvars?

Reply:
The goal of this PR is optimization, no? using localvars is objectively slower than a direct access for no real reason unless for some reason we need this in lua, at which point we'd write a function hook to that variable

Reply:
I'll defer to your judgement on that one. I might need some hand-holding to do what you're suggesting, though. Are you saying add a private member in `charentity.h` and save the timestamp there instead?

Reply:
Yeah, that's the ticket